native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java

```diff
@@ -67,13 +69,18 @@ public CocosSensorHandler(final Context context) {
     // Getter & Setter
     // ===========================================================
     public void enable() {
-        mSensorManager.registerListener(this, mAcceleration, mSamplingPeriodUs);
-        mSensorManager.registerListener(this, mAccelerationIncludingGravity, mSamplingPeriodUs);
-        mSensorManager.registerListener(this, mGyroscope, mSamplingPeriodUs);
+        if (mEnableSensor) {
+            mHasRegisterListener = true;
+            mSensorManager.registerListener(this, mAcceleration, mSamplingPeriodUs);
+            mSensorManager.registerListener(this, mAccelerationIncludingGravity, mSamplingPeriodUs);
+            mSensorManager.registerListener(this, mGyroscope, mSamplingPeriodUs);
+        }
     }
 
     public void disable() {
-        this.mSensorManager.unregisterListener(this);
+        if (mHasRegisterListener) {
```

Comment:
enable() is controlled by mEnableSensor, and this function is controlled by mHasRegisterListener. They are not consistent.

Reply:
App maybe crash, while disable sensor, but call unregisterListener. Need to be recheck.